src/PackageUrl.cs

```diff
@@ -184,7 +185,14 @@ public string ToJson()
         /// <exception cref="InvalidCastException">If the json string cannot be deserialized into a <see cref="PackageURL"/>.</exception>
         public static PackageURL FromJson(string json)
         {
-            return JsonConvert.DeserializeObject<PackageURL>(json) ?? throw new InvalidCastException();
+            try
+            {
+                return JsonSerializer.Deserialize<PackageURL>(json) ?? throw new JsonException("The deserialized PackageURL was null.");
+            }
+            catch (Exception e)
```

Comment:
I'd prefer not wrapping any exceptions that get caught with an InvalidCastException.  It's likely better to just let the original exception bubble up.